docker_publisher.py

```diff
@@ -0,0 +1,299 @@
+        return False
+
+    # Same input and output format -> just copy
+    if command_output.split(' ')[0] == command_input.split(' ')[0]:
+        shutil.copyfile(source, dest)
+        return True
+
+    ret = subprocess.call("%s '%s' | %s > '%s'" % (command_input, source, command_output, dest), shell=True)
+    return ret == 0
+
+
+class DockerImagePublisher:
+    """Base class for handling the publishing of docker images."""
+
+    def releasedDockerImageVersion(self):
+        """This function returns an identifier for the released docker
+        image's version."""
+        raise "pure virtual"
+
+    def releaseDockerImage(self, version, rootfile):
+        """This function publishes the docker image with root fs layer @rootfile.
+        After this was called, releasedDockerImageVersion must return @version."""
+        raise "pure virtual"
+
+
+class DockerPublishException(Exception):
+    pass
+
+
+class DockerImageFetcher:
+    """Base class for handling the acquiring of docker images."""
+
+    def currentVersion(self):
+        """This function returns the version of the latest available version
+        of the image for the product."""
+        raise "pure virtual"
+
+    def getDockerImage(self, callback):
+        """This function downloads the root fs layer and calls callback
+        with its path as argument."""
+        raise "pure virtual"
+
+
+class DockerFetchException(Exception):
+    pass
+
+
+class DockerImagePublisherGit(DockerImagePublisher):
+    def __init__(self, git_path, git_branch, path, arch):
+        """Initialize a DockerImagePublisherGit with:
+        @git_path: Path to the local git repo clone
+        @git_branch: Branch of the image
+        @path: Path to the directory within the branch which will contain:
+        - Dockerfile (including version in a comment)
+        - *.tar.xz: Images"""
+        self.git_path = git_path
+        self.git_branch = git_branch
+        self.path = path
+        self.arch = arch
+
+        self.git_call = ["git", "-C", self.git_path]
+
+        ret = subprocess.call(self.git_call + ["fetch", "origin"])
+        if ret != 0:
+            raise DockerFetchException("Could not fetch from origin")
+
+    def releasedDockerImageVersion(self):
+        # Read from git using cat-file to avoid expensive checkout.
+        args = self.git_call + ["cat-file", "--textconv", "origin/%s:%s/Dockerfile" % (self.git_branch, self.path)]
+        with subprocess.Popen(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE) as p:
+            version_regex = re.compile("^# Version: (.+)$")
+            for line in p.stdout:
+                match = version_regex.match(line.decode('utf-8').strip())
+                if match:
+                    return match.group(1)
+
+        return "0"
+
+    def generateDockerFile(self, version, filename):
+        """Return the contents of a docker file for the image"""
+        return """FROM scratch
+MAINTAINER Fabian Vogt <fvogt@suse.com>
```

Comment:
Is this right for official image?

I see https://github.com/openSUSE/docker-containers/blob/master/openSUSE-Tumbleweed/config.kiwi#L15 has:

```
SUSE Containers Team <containers@suse.com>
```

Reply:
Yes, the containers team moved the responsibility of those images to me.

I plan to move the kiwi description file away from git (already done for 15), so what's in there is not correct anymore anyway.

Reply:
move responsibility from a team to a single person? Doesn't sound right. [bus-factor](https://en.wikipedia.org/wiki/Bus_factor)

Reply:
It is. Good that I mostly travel by train :-)

It might make sense to use a more generic mail address there, maybe `opensuse-factory@opensuse.org`?

Reply:
I guess

Reply:
Something more generic would definitely seem better.

Reply:
Seems like this still needs to be addressed. Seems like Factory is best if not perhaps opensuse-releaseteam@o.o?